services/src/main/java/org/keycloak/protocol/ciba/endpoints/BackchannelAuthenticationEndpoint.java

```diff
@@ -124,12 +136,13 @@ public Response processGrantRequest() {
         if (provider == null) {
             throw new RuntimeException("CIBA Decoupled Authentication Provider not setup properly.");
         }
-        provider.doBackchannelAuthentication(client, request, policy.getExpiresIn(), authResultId, userSessionIdWillBeCreated);
+        provider.doBackchannelAuthentication(client, request, expireIn, authResultId, userSessionIdWillBeCreated);
 
         ObjectNode response = JsonSerialization.createObjectNode();
         response.put(CIBAConstants.AUTH_REQ_ID, authReqId);
-        response.put(CIBAConstants.EXPIRES_IN, policy.getExpiresIn());
-        if (policy.getInterval() > 0) response.put(CIBAConstants.INTERVAL, policy.getInterval());
+
+        response.put(CIBAConstants.EXPIRES_IN, expireIn);
+        if (policy.getInterval() > 0) response.put(CIBAConstants.INTERVAL, interval);
```

Comment:
IMO, how about the following?
```
        if (interval > 0) response.put(CIBAConstants.INTERVAL, interval);
```

Reply:
yes, right. I missed it. I'll fix it. thank you